test/ccapi/unittest_ccapi.cpp

```diff
@@ -147,6 +147,7 @@ TEST(ccapi_optimizer, construct_01_n) {
 TEST(ccapi_optimizer, construct_02_p) {
   EXPECT_NO_THROW(ml::train::optimizer::Adam());
   EXPECT_NO_THROW(ml::train::optimizer::SGD());
+  EXPECT_NO_THROW(ml::train::optimizer::AdamW());
```

Comment:
Have you ever conducted any actual training tests other than ccapi construct test?
when i tested on MNIST Application Example i got  ```segmentation fault (core dumped)``` error

tested on this code
```.cpp
  std::unique_ptr<ml::train::Model> model;
  model = createModel(ml::train::ModelType::NEURAL_NET);
  model->load(config, ml::train::ModelFormat::MODEL_FORMAT_INI_WITH_BIN);
  model->compile();
  model->initialize();
  model->setDataset(ml::train::DatasetModeType::MODE_TRAIN, dataset_train);
  model->setDataset(ml::train::DatasetModeType::MODE_VALID, dataset_val);
  model->train();
```
with this ```ini``` file
```.ini
# Network Section : Network

[Model]
Type = NeuralNetwork                # Network Type : Regression, KNN, NeuralNetwork
Epochs = 1500                       # Epochs
Loss = cross                        # Loss function : mse (mean squared error)
                                    # cross ( for cross entropy )
# Save_Path = "mnist_model.bin"     # model path to save / read
batch_size = 32                     # batch size

[Optimizer]
Type = adamw
beta1 = 0.9                         # beta 1 for adam
beta2 = 0.999                       # beta 2 for adam
epsilon = 1e-7                      # epsilon for adam

# Layer Section : Name
[inputlayer]
Type = input
Input_Shape = 1:28:28

# Layer Section : Name
[conv2d_c1_layer]
Type = conv2d
input_layers = inputlayer
kernel_size = 5,5
bias_initializer=zeros
Activation=sigmoid
weight_initializer = xavier_uniform
filters = 6
stride = 1,1
padding = 0,0

[pooling2d_p1]
Type=pooling2d
input_layers = conv2d_c1_layer
pool_size = 2,2
stride =2,2
padding = 0,0
pooling = average

[conv2d_c2_layer]
Type = conv2d
input_layers = pooling2d_p1
kernel_size = 5,5
bias_initializer=zeros
Activation=sigmoid
weight_initializer = xavier_uniform
filters = 12
stride = 1,1
padding = 0,0

[pooling2d_p2]
Type=pooling2d
input_layers = conv2d_c2_layer
pool_size = 2,2
stride =2,2
padding = 0,0
pooling = average

[flatten]
Type=flatten
input_layers = pooling2d_p2

[outputlayer]
Type = fully_connected
input_layers=flatten
Unit = 10                           # Output Layer Dimension ( = Weight Width )
weight_initializer = xavier_uniform
bias_initializer = zeros
Activation = softmax                # activation : sigmoid, softmax
```

Reply:
The code should now actually work.

Thanks.